### DISCUSSION TITLE
Is useAnimatedGestureHandler only supported for PanGestureHandler?

### DISCUSSION THREAD

**Discussion by evrimfeyyaz** [software-mansion/react-native-reanimated#1931] [Q&A] [answered]
I'm trying to use `useAnimatedGestureHandler` with `LongPressGestureHandler` and it doesn't work. None of the handlers get called. Also, the return value of `useAnimatedGestureHandler` is `OnGestureEvent<PanGestureHandlerGestureEvent>`.

Does `useAnimatedGestureHandler` only work with `PanGestureHandler`? Is there any specific reason why it is this way? And is there a workaround (apart from implementing long-press detection using `PanGestureHandler`)?

**Comment by evrimfeyyaz** [comment, marked as answer]
I changed the child component of the `LongPressGestureHandler` from `View` to `Animated.View` and it's working now. Although, the type signature is still wrong and TypeScript is complaining.

**Comment by jgonet** [comment]
`useAnimatedGestureHandler` uses PanGestureHandler [as a default](https://github.com/software-mansion/react-native-reanimated/blob/master/react-native-reanimated.d.ts#L509-L515). You can parametrize it by using other handler, like `useAnimatedGestureHandler<TapGestureHandlerGestureEvent>`.

**Reply by evrimfeyyaz** [reply]
You are completely right, I missed that. Thanks, @jakub-gonet.
